metricbeat/module/kibana/stats/stats.go

```diff
@@ -63,22 +67,22 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 		return nil, err
 	}
 
-	http, err := helper.NewHTTP(base)
+	statsHTTP, err := helper.NewHTTP(base)
 	if err != nil {
 		return nil, err
 	}
 
-	kibanaVersion, err := kibana.GetVersion(http, statsPath)
+	kibanaVersion, err := kibana.GetVersion(statsHTTP, statsPath)
```

Comment:
One thing I just realised that bugs me a bit with this implementation is that it requires Kibana to be up and running normally when Metricbeat is started. Assuming Kibana is flaky during the moment the check happens, the metricset will not be running but error out even though Kibana could be totally fine again 1 sec later.

Not for this PR but thought worth mentioning.

Reply:
Good point. I suppose we could move this check (and the code dependent on it) into `Fetch` but that might be too expensive to run every fetch cycle?

Reply:
Maybe we need another metricset lifecycle method in addition to `New()` and `Fetch()`, say `Ping()` (I know, not a great name), which runs periodically but less frequently than `Fetch()`. The period can be defined per-metricset in that metricset's `New()` function. The code in `Ping()` can add/overwrite info in the metricset's struct so the code in `Fetch()` can use it. Thoughts?

Reply:
What about using fetch but do it only once? I remember we do something like that in other cases.

Reply:
Merged as it's something we should discuss as a follow up.